src/dbos-executor.ts

```diff
@@ -1823,12 +1843,17 @@ export class DBOSExecutor implements DBOSExecutorContext {
 
     const handlerArray: WorkflowHandle<unknown>[] = [];
     for (const execID of executorIDs) {
-      if (execID === 'local' && process.env.DBOS__VMID) {
-        this.logger.debug(`Skip local recovery because it's running in a VM: ${process.env.DBOS__VMID}`);
-        continue;
-      }
       this.logger.debug(`Recovering workflows assigned to executor: ${execID}`);
-      const pendingWorkflows = await this.systemDatabase.getPendingWorkflows(execID);
+      const { numWrongVersion, pendingWorkflows } = await this.systemDatabase.getPendingWorkflows(
+        execID,
+        globalAppVersion.version,
+      );
+      if (pendingWorkflows.length > 0) {
+        this.logger.info(`Recovering ${pendingWorkflows.length} workflows`);
+      }
+      if (numWrongVersion > 0) {
+        this.logger.info(`${numWrongVersion} workflows with a different app version were not recovered`);
```

Comment:
I think this message could be confusing. Especially if we're going to print it every time we start a VM.